components/MatterInterface/MatterInterface.cpp

```diff
@@ -128,58 +173,23 @@ bool Matter_TaskInit(MatterInterfaceHandler_t *MatterInterfaceHandler)
         Log_RamOccupy("Matter", "making node");
         node::config_t node_config;
         node_t *node = node::create(&node_config, app_attribute_update_cb, app_identification_cb);
-        Log_RamOccupy("Matter", "making node");
-        Log_RamOccupy("Matter", "making endpoint");
-        on_off_switch::config_t switch_config;
-        endpoint_t *endpoint1 = on_off_switch::create(node, &switch_config, ENDPOINT_FLAG_NONE, switch_handle);
-
-        Log_RamOccupy("Matter", "making endpoint");
-        done_coffee_maker::config_t coffee_maker_config;
-        endpoint_t *endpoint2 = done_coffee_maker::create(node, &coffee_maker_config, ENDPOINT_FLAG_NONE, NULL /*coffee_maker_handle*/);
-
-        // "Temperature Measurement", "Refrigerator and Temperature Controlled Cabinet Mode Select" are optional cluster for temperature_controlled_cabinet device type so we are not adding them by default.
-        // temperature_controlled_cabinet::config_t temperature_controlled_cabinet_config;
-        // endpoint_t *endpoint3 = temperature_controlled_cabinet::create(node, &temperature_controlled_cabinet_config, ENDPOINT_FLAG_NONE, NULL);
-
-        /* These node and endpoint handles can be used to create/add other endpoints and clusters. */
-        if (!node || !endpoint1 || !endpoint2 /*|| !endpoint3*/)
+        Log_RamOccupy("Matter", "making node");        
+        if (!node)
         {
             ESP_LOGE(TAG, "Matter node creation failed");
-        }
-
-        // SEZ@Done on_off_switch doesn't have group user by default.
-        // so add it manually.
-
-        /* Add group cluster to the switch endpoint */
-        cluster::groups::config_t groups_config;
-        cluster::groups::create(endpoint1, &groups_config, CLUSTER_FLAG_SERVER | CLUSTER_FLAG_CLIENT);
-
-        switch_endpoint_id = endpoint::get_id(endpoint1);
-        ESP_LOGI(TAG, "Switch created with endpoint_id %d", switch_endpoint_id);
-
-        coffee_maker_endpoint_id = endpoint::get_id(endpoint2);
-        ESP_LOGI(TAG, "Coffee Maker created with endpoint_id %d", coffee_maker_endpoint_id);
-
-        // esp_matter::cluster_t *cluster = esp_matter::cluster::get(endpoint1, chip::app::Clusters::TemperatureControl::Id);
-
-        // // Atleast one of temperature_number and temperature_level feature is mandatory.
-        // cluster::temperature_control::feature::temperature_number::config_t temperature_number_config;
-        // cluster::temperature_control::feature::temperature_number::add(cluster, &temperature_number_config);
-
-        // temp_ctrl_endpoint_id = endpoint::get_id(endpoint3);
-        // ESP_LOGI(TAG, "Temperature controlled cabinet created with endpoint_id %d", temp_ctrl_endpoint_id);
+        }         
 
+#ifdef CONFIG_DONE_MATTER_DEVICE_COFFEE_MAKER            
+        Log_RamOccupy("Matter", "making endpoint");        
+        err = DoneCoffeeMakerCreate(node);        
+#endif                   
```

Comment:
please check you error,you just receive it!

Reply:
edit every them in DoneMatterCoffeMaker.cpp/.h.but not in  DoneMatterEndpoint and Interface, because defined by Matter SDK.